wellcomeml/datasets/winer.py

```diff
@@ -0,0 +1,271 @@
+# WiNER: A Wikipedia Annotated Corpus for Named Entity Recognition
+# https://www.aclweb.org/anthology/I17-1042/
+
+import tarfile
+import os
+from tqdm import tqdm 
+import json
+import random
+
+from wellcomeml.datasets.download import check_cache_and_download
+from wellcomeml.logger import logger
+
+def create_train_test(
+    NE_path, vocab_path, docs_path,
+    train_processed_path, test_processed_path,
+    n_sample, prop_train, rand_seed=42
+    ):
+    """
+    Function to transform the raw WiNER datasets into training
+    and testing datasets suitable for training a NER model.
+    This NE_path file gives the entities, but they need to be linked
+    to the sentences they came from, which involves translating
+    them from the vectorised form in the docs_path file
+    using the vocab_path file.
+
+    NE_path: File path to the WiNER raw data named entities
+    vocab_path: File path to the WiNER raw data vocab dictionary
+    docs_path: File path to the WiNER raw data documents texts
+            (words coded according to the vocab)
+    train_processed_path: Output file path for the training data
+    test_processed_path: Output file path for the testing data
+    n_sample: Number of files to get data from, each file contains about 1000 documents
+    prop_train: Proportion of the file sample that will be in the training data
+
+    Utilise the three datasources to get the entities from a sample
+    of documents and save to a training and testing dataset files
+    in the form:
+
+    ID 12345
+
+    John Smith PERSON
+    rejects O
+    German LOC
+    call O
+    
+    Next 0
+    sentence 0
+
+    ID 444
+
+    Next
+    document
+    """
+
+    # Set the random seed for taking a sample
+    random.seed(rand_seed)
+    n_train = round(prop_train*n_sample)
+
+    # Make a dictionary for the entities for every article
+    # in the form
+    # entities = {article_id: {sentence_id:[[0, 4, 0], [18, 25, 1]]}}
+    # The sentence id is which number sentence (e.g. 1st, 2nd) the entities
+    # are from in the whole document
+
+    logger.info("Creating entities dictionary from {}".format(NE_path))
+    entities = {}
+    with tarfile.open(NE_path, "r:bz2") as tar:
+        for member in tqdm(tar.getmembers()):
+            f = tar.extractfile(member)
+            if f:
```

Comment:
to avoid nesting everything below once more you can just do
```
if f:
    continue
```
a comment as to when this happen might be useful as well